packages/orchestration/src/orchestration.js

```diff
@@ -203,16 +227,25 @@ const prepareOrchestration = (zone, createChainAccount) =>
           powers.init(/** @type {keyof OrchestrationPowers} */ (name), power);
         }
       }
-      return { powers, icaControllerNonce: 0 };
+      return {
+        powers,
+        queryPort: undefined,
+        icqControllerNonce: 0,
+        icaControllerNonce: 0,
+      };
     },
     {
       self: {
-        async bindPort() {
+        /**
+         * @param {{portType: string; nonceKey: string;}} opts
+         */
+        async bindPort(opts) {
           const network = getPower(this.state.powers, 'network');
+          const nonce = this.state[opts.nonceKey];
           const port = await E(network).bindPort(
-            `/ibc-port/icacontroller-${this.state.icaControllerNonce}`,
+            `/ibc-port/${opts.portType}-${nonce}`,
           );
-          this.state.icaControllerNonce += 1;
+          this.state[opts.nonceKey] += 1;
```

Comment:
should this be before the remote call? the remote might throw with partial failure and a subsequent request would use the same value. cc @iomekam

Reply:
_I am going to rebase this PR on top of #9228 once ready, so please disregard this logic. _

Was not thinking of the partial failure case, although it seems harmless as the object containing `Port` is not returned to the user until the Port + Connection are successfully instantiated. Nonetheless, this seems like it wouldn't comply with the Checks Effects Interactions pattern and it's better to make the state update before the remote call.